wire/msg.go

```diff
@@ -42,40 +53,35 @@ type (
 		// Msg contained in this Envelope. Not embedded so Envelope doesn't implement Msg.
 		Msg Msg
 	}
-)
 
-// Encode encodes an Envelope into an io.Writer.
-func (env *Envelope) Encode(w io.Writer) error {
```

Comment:
The Envelope.Encode function should not be removed. It still has its place for supporting perunio encoding, right? Same holds for the Decode function.

Reply:
Nope, they are not required. Because, the `Encode` and `Decode` methods are replaced by `EncodeEnvelope` and `DecodeEnvelope` functions. Because, these two functions are using a package level singleton instance for encoding/decoding.

This was also suggested as a better approach in the design proposal concerning wire serialization abstraction. It is mentioned in the [Implementation](https://github.com/hyperledger-labs/perun-proposals/blob/main/design/005-Wire-Encoding-Abstraction.md#implementation) section.